### PR TITLE
chore: In DeleteAccount, set activeAccount nil if deleting the active account

### PR DESCRIPTION
**Pull request by jefft0**
This is a follow-on to DeleteAccount support in PR #43.

### CODE REVIEW

File: service/api.go [hunk 1]
```diff
@@ -184,6 +184,11 @@ func (s *gnomobileService) DeleteAccount(ctx context.Context, req *connect.Reque
 			return nil, err
 		}
 	}
+	if s.activeAccount != nil &&
+		(s.activeAccount.GetName() == req.Msg.NameOrBech32 || crypto.AddressToBech32(s.activeAccount.GetAddress()) == req.Msg.NameOrBech32) {
+		// The deleted account was the active account.
+		s.activeAccount = nil
```
Comment (D4ryl00) [inline service/api.go:190]:
```suggestion
        s.lock.Lock()
		s.activeAccount = nil
		s.lock.Unlock()
```

**Review by D4ryl00 — CHANGES_REQUESTED** [review]
You should protect the write by the mutex.

File: service/api.go (hunk 1 above)
Comment (jefft0) [reply service/api.go:191]: Yes, thank you. I did a force push to fix it.

**Review by D4ryl00 — APPROVED** [review]
LGTM :+1:

**Pull request merged**